hudi-client/src/main/java/org/apache/hudi/HoodieWriteClient.java

```diff
@@ -325,6 +326,31 @@ public static SparkConf registerClasses(SparkConf conf) {
     }
   }
 
+  /**
+   * Deletes a bunch of keys from the Hoodie table, at the supplied commitTime
+   */
+  public JavaRDD<WriteStatus> delete(JavaRDD<HoodieKey> keys, final String commitTime) {
+    HoodieTable<T> table = getTableAndInitCtx();
+    try {
+      // De-dupe/merge if needed
+      JavaRDD<HoodieKey> dedupedKeys =
+          combineKeysOnCondition(config.shouldCombineBeforeUpsert(), keys, config.getUpsertShuffleParallelism());
```

Comment:
Introduce a new config config.shouldCombineBeforeDelete() and config.getDeleteShuffleParallelism()?

Reply:
@bvaradar just for my understanding: If we pre-combine for upsert, we also do pre-combine for other write actions? Is it possible either now or in future that we would want to pre-combine for only insert/upsert and not for delete?

Reply:
@bhasudha : We do support precombine for upsert, insert and bulkInsert calls. For delete, it is merely just deduping keys - there isn't any custom combine logic as the payload is not there.  Yeah, I thought Its ok to have similar semantics for deletes as other APIs just to be consistent